controlplane/kubeadm/internal/etcd/etcd.go

```diff
@@ -169,15 +169,18 @@ func NewClient(ctx context.Context, config ClientConfiguration) (*Client, error)
 	client, err := newEtcdClient(ctx, etcdClient, callTimeout)
 	if err != nil {
 		closeErr := etcdClient.Close()
-		return nil, errors.Wrap(kerrors.NewAggregate([]error{err, closeErr}), "unable to create etcd client")
+		if closeErr != nil {
+			return nil, err
+		}
```

Comment:
```suggestion
		closeErr := etcdClient.Close()
```

NewAggregate already drops if errors are nil so can directly use that. Otherwise: should the if be the other way around?